### PR TITLE
add gen2 function docs

### PR DESCRIPTION
**Pull request by josefaidt**
#### Description of changes:

image

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/build-a-backend/functions/additional-configuration/index.mdx [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{/* // import { getChildPageNodes } from '@/utils/getChildPageNodes'; */}
+
+export const meta = {
+  title: 'Additional configuration',
```
Comment (renebrandel) [inline src/pages/gen2/build-a-backend/functions/additional-configuration/index.mdx:4]: Resist titles like "additional" or "advanced" they become catch all buckets with limited meaning for the customer.

File: src/pages/gen2/build-a-backend/functions/extend-with-cdk/index.mdx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export const meta = {
+  title: 'Extend with CDK',
```
Comment (renebrandel) [inline src/pages/gen2/build-a-backend/functions/extend-with-cdk/index.mdx:2]: Modify Amplify-generated Lambda resources with CDK

File: src/pages/gen2/build-a-backend/functions/extend-with-cdk/index.mdx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+export function getStaticProps() {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+## Adding IAM Policies
```
Comment (renebrandel) [inline src/pages/gen2/build-a-backend/functions/extend-with-cdk/index.mdx:15]: Does this title not better belong in the access rules section?

File: src/pages/gen2/build-a-backend/functions/extend-with-cdk/index.mdx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+{/* ## Lambda Layers */}
```
Comment (renebrandel) [inline src/pages/gen2/build-a-backend/functions/extend-with-cdk/index.mdx:17]: Shouldn't this be a top level page? Whether it's done via CDK or not is secondary.

**Review by renebrandel — CHANGES_REQUESTED** [review]
Requested changes on the titles.

File: src/pages/gen2/build-a-backend/functions/extend-with-cdk/index.mdx (hunk 3 above)
Comment (josefaidt) [reply src/pages/gen2/build-a-backend/functions/extend-with-cdk/index.mdx:15]: this is for IAM policies beyond the access prop, but I plan to call it out in the access rules section. mostly a note of circular dependencies

**Pull request closed without merging**